src/CoxGrpRec.jl

```diff
@@ -53,6 +53,81 @@ function coxeter_group_recursive(M::Matrix{Int64})
+
+julia> CG
+Coxeter Group with Coxeter Matrix:
+3×3 Matrix{Int64}:
+ 1  3  2
+ 3  1  3
+ 2  3  1
+and generating Set:
+["<1>", "<2>", "<3>"]
+
+```
+```julia-repl
+julia> CG, _ = coxeter_group_recursive("A3","B4")
+[1 3 2; 3 1 3; 2 3 1]
+[1 3 2 2; 3 1 3 2; 2 3 1 4; 2 2 4 1]
+(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), CoxEltRec[CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [1]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [2]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [3]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [4]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [5]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [6]), CoxEltRec(CoxGrpRec([1 3 … 2 2; 3 1 … 2 2; … ; 2 2 … 1 4; 2 2 … 4 1], ["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]), [7])])
+
+julia> CG
+Coxeter Group with Coxeter Matrix:
+7×7 Matrix{Int64}:
+ 1  3  2  2  2  2  2
+ 3  1  3  2  2  2  2
+ 2  3  1  2  2  2  2
+ 2  2  2  1  3  2  2
+ 2  2  2  3  1  3  2
+ 2  2  2  2  3  1  4
+ 2  2  2  2  2  4  1
+and generating Set:
+["<1>", "<2>", "<3>", "<4>", "<5>", "<6>", "<7>"]
+
+```
+```julia-repl
+julia> CG, _ = coxeter_group_recursive("D~4")
+(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), CoxEltRec[CoxEltRec(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), [1]), CoxEltRec(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), [2]), CoxEltRec(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), [3]), CoxEltRec(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), [4]), CoxEltRec(CoxGrpRec([1 2 … 2 2; 2 1 … 2 2; … ; 2 2 … 1 2; 2 2 … 2 1], ["<1>", "<2>", "<3>", "<4>", "<5>"]), [5])])
+
+julia> CG
+Coxeter Group with Coxeter Matrix:
+5×5 Matrix{Int64}:
+ 1  2  3  2  2
+ 2  1  3  2  2
+ 3  3  1  3  3
+ 2  2  3  1  2
+ 2  2  3  2  1
+and generating Set:
+["<1>", "<2>", "<3>", "<4>", "<5>"]
+
+```
+"""
+function coxeter_group_recursive(groupType::String ...)
+    return coxeter_group_recursive([s for s in groupType])
+end
+
+function coxeter_group_recursive(S::Array{String})
+    matrix_array = Array{Matrix{Int64},1}(undef,length(S))
+    for i in 1:length(S)
+        matrix_array[i] = coxeter_matrix_from_group_type(S[i])
+    end
+    M = matrix_array[1]
```

Comment:
Maybe add a comment that the purpose of the next few lines is to make a "block-diagonal" matrix, but where the zeros are actually 2's since it's a Coxeter matrix. I wonder if this functionality should really be in the other piece of code that returns Coxeter matrices.